### ISSUE TITLE
Opening workbooks changes file name to lower case

### ISSUE THREAD

**Issue by qzyu999** [xlwings/xlwings#2052]
#### OS (e.g. Windows 10 or macOS Sierra)
Windows 10
#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
xlwings 0.15.4 Python 2.7
#### Describe your issue (incl. Traceback!)
I didn't have this issue before, but all of a sudden when I run my script to open a Excel workbook using xw.Book the filename is changed to all lowercase. I tried to open the same file simply using the xw.Book command and the same result happens. Not sure what's causing the filename to be automatically changed to all lower case. I saw the issue [here](https://stackoverflow.com/questions/71389786/xlwings-book-open-excel-workbook-but-change-the-file-name-to-lowercase), but the solution listed is not too useful for me.
```python
xw.Book(os.path.join(FILE_PATH, 'FILENAME.xlsm'))
```

#### Include a minimal code sample to reproduce the issue (and attach a sample workbook if required!)

```python
xw.Book(os.path.join(FILE_PATH, 'FILENAME.xlsm'))
```

**Comment by fzumstein** [comment]
I don't think there's anything that changed, I believe it's been there forever, but it usually shouldn't cause too much issues as Windows is case-insensitive. However, I agree, it should maintain the proper capitalization in:

```python
book = xw.Book(...)
book.name
```

**Comment by jakubrojcek** [comment]
I have this issue as well. It's unexpected.

**Comment by ediouslo** [comment]
I have the same issue as well. As a result, I am getting a FileNotFoundError...
Do you know any solution to work around this?

**Comment by fzumstein** [comment]
@ediouslo this specific issue shouldn't cause this error. Try to turn your file name into lowercase and let me know should that resolve the issue.

**Comment by fzumstein** [comment]
I could only replicate this on macOS, not Windows, but is now fixed in 0.28.9

**Comment by ozgurkalan** [comment]
> 

I confirm that the issue is solved. 
I am working on a windows 11 PC with Anaconda running. Upgrading solved weird character lowercase confusion with Turkish chars.
